networkx/algorithms/centrality/laplacian.py

```diff
@@ -0,0 +1,182 @@
+"""Laplacian centrality measures."""
+
+import networkx as nx
+import numpy as np
+from scipy import linalg
+import scipy as sp
+from scipy import sparse
+
+__all__ = ["laplacian_centrality"]
+
+
+def delete_row_csr(mat, i):
+    r"""Delete specific row from scipy matrix.
+
+    Delete specific row from a scipy csr matrix based on the index.
+
+    Parameters
+    ----------
+    mat : sparse matrix
+
+    i : index of the column to drop
+
+
+    Returns
+    -------
+    csr matrxix : sparse matrix
+
+    """
+    if not isinstance(mat, sp.sparse.csr_matrix):
+        raise ValueError("works only for CSR format -- use .tocsr() first")
+    n = mat.indptr[i + 1] - mat.indptr[i]
+    if n > 0:
+        mat.data[mat.indptr[i] : -n] = mat.data[mat.indptr[i + 1] :]
+        mat.data = mat.data[:-n]
+        mat.indices[mat.indptr[i] : -n] = mat.indices[mat.indptr[i + 1] :]
+        mat.indices = mat.indices[:-n]
+    mat.indptr[i:-1] = mat.indptr[i + 1 :]
+    mat.indptr[i:] -= n
+    mat.indptr = mat.indptr[:-1]
+    mat._shape = (mat._shape[0] - 1, mat._shape[1])
+
+    return mat
+
+
+def delete_col_csr(mat, i):
+    r"""Delete specific col from scipy matrix.
+
+    Delete specific column from a scipy csr matrix based on the index.
+
+    Parameters
+    ----------
+    mat : sparse matrix
+
+    i : index of the column to drop
+
+    Returns
+    -------
+    csr matrxix : sparse matrix
+
+    """
+    idx_to_drop = np.unique(i)
+    C = mat.tocoo()
+    keep = ~np.in1d(C.col, idx_to_drop)
+    C.data, C.row, C.col = C.data[keep], C.row[keep], C.col[keep]
+    C.col -= idx_to_drop.searchsorted(C.col)  # decrement column indices
+    C._shape = (C.shape[0], C.shape[1] - len(idx_to_drop))
+    return C.tocsr()
+
+    # @not_implemented_for("multigraph")
+
+
+def laplacian_centrality(
+    G, normalized=True, nbunch=None, directed_laplacian_matrix_args=None
+):
+    r"""Compute the Laplacian centrality for nodes in the graph `G`.
+
+    The Laplacian Centrality of a node `i` is measured by the drop in the Laplacian Energy
+    after deleting node `i` from the graph.
+
+    .. math::
+
+        C_L(u_i,G) = \frac{(\Delta E)_i}{E_L (G)} = \frac{E_L (G)-E_L (G_i)}{E_L (G)}
+
+        E_L (G) = \sum_{i=0}^n \lambda_i^2
+
+    Where $E_L (G)$ is the Laplacian energy of graph `G`,
+    E_L (G_i) is the Laplacian energy of graph `G` after deleting node `i`
+    and $\lambda_i$ are the eigenvalues of its Laplacian matrix.
+
+    Parameters
+    ----------
+
+    G : graph
+        A networkx graph
+
+    normalized : bool (default = True)
+        If False then the algorithm does't calculate Laplacian energy of entire graph `G`,
+        just the Laplacian energy of the given nodes.
```

Comment:
?? I think this is a description of the `nbunch` parameter. Should this be something about normalizing the centrality measure?

Reply:
I wrote this here: 

        If True the Laplacian energy of a given node is divided by the Laplacian energy of the entire graph.
        If False the algorithm returns the Laplacian energy of a given node.